src/interfaces.ts

```diff
@@ -168,3 +177,20 @@ export interface ISendLogRequest {
   };
   traces: ITrace[];
 }
+
+export interface ILogMessageOptions {
+  skipDefaultFormatter?: boolean,
+  skipServer?: boolean,
+  skipSecondaryLogger?: boolean,
+}
+
+// TODO: is there a better way to declare this?
```

Comment:
I'm fine with this. I wonder if you could create a "combined" type that uses intersects both of these so you don't have to union them everywhere else. Something like: 

``` ts
export type NextFn = NextFnWithoutParams & NextFnWithParams;
export type NextFnWithoutParams = () => void;
export type NextFnWithParams = (level: LogLevel, message: string | Error, details?: any, options?: ILogMessageOptions) => void;
```
Then you can just declare `NextFn` as the type. TS should be smart enough that if you start passing in params, it will use the `NextFnWithParams` type. 
``` ts
  private defaultFormatter = (
    logLevel: LogLevel,
    message: string | Error,
    details: any | undefined,
    messageOptions: ILogMessageOptions,
    next: NextFn // <- just use one
  ): void => { ... }
```